lib/Net/OAuth2/AuthorizationServer/Defaults.pm

```diff
@@ -295,7 +295,7 @@ sub _verify_access_token_jwt {
             }
         }
 
-        return ( $access_token_payload, undef );
+        return ( $access_token_payload->{client}, undef, $access_token_payload->{scopes} );
```

Comment:
Can you explain this change a little more? There's now an extra argument coming back from `_verify_access_token_jwt` but i don't see what this fixes, also some tests for this might be useful.

Reply:
Actually I see - brings this method in line with the `_verify_access_token` method. Although I don't see the reason for the additional return value?

Reply:
Right now, the refresh token of the mojolicious plugin does not work because of that, on the mojo plugin, in the sub _access_token_request, the _verifiy_credentials is called:

```perl
  my ( $client,$error,$scope,$user_id,$old_refresh_token ) = _verify_credentials(
    $self,$Grant,$grant_type,$refresh_token,$client_id,$client_secret,
    $auth_code,$username,$password,$uri
  );
```

that one in turn call _verify_access_token which call _verify_access_token_jwt

Here in client, instead of returning the client, we return the full token payload, and nothing in scope, then in turn in _access_token_request, new access/refresh token are generated by:

```perl
    my $access_token  = $Grant->token(
      client_id => $client,
      scopes    => $scope,
      type      => 'access',
      user_id   => $user_id,
      jwt_claims_cb => $JWTCallback,
    );

    my $refresh_token  = $Grant->token(
      client_id => $client,
      scopes    => $scope,
      type      => 'refresh',
      user_id   => $user_id,
      jwt_claims_cb => $JWTCallback,
    );
```
As with jwt, client and scope are not correct, the generated token is of the following structure:

```perl
$VAR1 = {
          'sub' => 'not the passed user_id',
          'client' => {
                      'iat' => 1502030392,
                      'scopes' => [
                                  'betrayal'
                                ],
                      'user_id' => undef,
                      'sub' => 'not the passed user_id',
                      'client' => 'frey',
                      'iss' => 'some iss',
                      'jti' => 'b9uevCxXOUoxKYNiYYBLSRcfy1lyIpdI',
                      'type' => 'refresh',
                      'aud' => undef
                    },
          'aud' => undef,
          'type' => 'access',
          'jti' => 'Md2YeQTqj3HNRiDyvg1f5BjMJKrhMvRV',
          'iss' => 'some iss',
          'iat' => 1502030392,
          'user_id' => undef,
          'scopes' => undef,
          'exp' => 1502033992
        };
```
That merge request is linked as well to https://github.com/Humanstate/mojolicious-plugin-oauth2-server/issues/18

Hope that clarifies

Reply:
OK, understood. I'll try to get this merged and released in the next couple of days.

Reply:
thanks